### PR TITLE
add two tabs to Table S1

### PR DESCRIPTION
**Pull request by jharenza**
#### What scientific question is your analysis addressing?
Taking tables out of methods and adding to Table S1

#### What was your approach?
tribbling

#### What GitHub issue does your pull request address?
https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/1319

#### Which areas should receive a particularly close look?
NA

#### Is there anything that you want to discuss further?
no

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
yes

#### What is your summary of the results?
two tabs added

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Review by sjspielman — CHANGES_REQUESTED** [review]
Something I'm not clear on is why some columns are documented as possible values of "text" but others are not. At first I thought it had to do with length of the possible values, but then I see for example `ethnicity` is text. I put a suggestion in for how populating the possible values can be done programmatically too.

File: tables/output_tables.Rmd [hunk 1]
```diff
@@ -76,18 +76,85 @@ histology_df %>%
...
+"sample_type","Broad sample type","Normal;Tumor",
+"seq_center","Sequencing center","BGI;BGI@CHOP Genome Center;Genomic Clinical Core at Sidra Medical and Research Center;NantOmics;The Translational Genomics Research Institute",
+"short_histology","Abbreviated `integrated_diagnosis` or `broad_histology` for plotting purposes","text",
+"tumor_descriptor","Phase of therapy from which tumor was derived","Initial CNS Tumor;Progressive Progressive Disease Post-Mortem;Recurrence;Second Malignancy;Unavailable",
+"tumor_fraction","Theta2 tumor DNA fraction estimate","numeric",
+"tumor_ploidy","Control-FREEC ploidy","numeric"
+)
+
+# Sheet 2: Define array of samples that need to be classified 
 to_be_classified_samples <- c("BS_3J4T2YYW", "BS_7F07M7JG", "BS_AG0BZN6F", "BS_HZNKSQ17", "BS_V6HKXX1Z", "BS_WP9J88EB", "BS_YYAPSA5P")
```
Comment (sjspielman) [inline tables/output_tables.Rmd:125]: Can we add some commenting here for where these samples come from?

File: tables/output_tables.Rmd [hunk 2]
```diff
@@ -76,18 +76,85 @@ histology_df %>%
...
     Kids_First_Biospecimen_ID %in% to_be_classified_samples,
     "MB, To be classified",
     molecular_subtype
-  )) %>%
-  # export with argument `keepNA=TRUE` so NAs come up as NAs and not blank cells
-  write.xlsx(table_s1_file, overwrite=TRUE, keepNA=TRUE)
+  ))
+
+# Sheet 3: CNS region definition
+cns_regions <- tribble(
```
Comment (sjspielman) [inline tables/output_tables.Rmd:136]: Add some more comments here too for where this information came from.

File: tables/README.md [hunk 3]
```diff
@@ -26,5 +29,11 @@ Table S3: RNA results table
     - Sheet 3: this tables shows the fractions of for various immune cell types in the tumor microenvironment (TME) for each RNA sample as calculated with [`quanTIseq`](https://doi.org/10.1186/s13073-019-0638-6).
         - Input files are from module [immune-deconv](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/immune-deconv)
     
-    
+Table S4: RNA results table
```
[the remark this answers is not in the document]
Comment (jharenza) [reply tables/README.md:32]: ha, copy error

File: tables/output_tables.Rmd [hunk 4]
```diff
@@ -76,18 +76,85 @@ histology_df %>%
...
+
+readme <- tribble(
+~`Histology column`,~Definition,~`Possible values`,
+"age_at_diagnosis_days","Patient age at diagnosis in days","numeric",
+"age_last_update_days","Patient age at the last clinical event/update in days","numeric",
+"aliquot_id","External aliquot identifier","variable",
+"broad_histology","Broad WHO 2016 classification of cancer type","text",
+"cancer_group","Harmonized cancer groupings for plots","text",
+"cancer_predispositions","Reported cancer predisposition syndromes","text",
+"CNS_region","Harmonized brain region based on `primary_site`","Hemispheric;Midline;Mixed;Optic
```
[the remark this answers is not in the document]
Comment (jharenza) [reply tables/output_tables.Rmd:89]: good idea!

**Comment by jharenza** [comment]
@sjspielman made the updates - just had a question as to whether we want to keep the programmatic option of all cancer groups there since it is quite long. I suppose people can wrap that column, so may be of interest.

File: tables/output_tables.Rmd [hunk 5]
```diff
@@ -76,18 +76,86 @@ histology_df %>%
...
+"sample_id","External biospecimen identifier","alphanumeric",
+"sample_type","Broad sample type",paste(unique(histology_df$sample_type), collapse = "; "),
+"seq_center","Sequencing center",paste(unique(histology_df$seq_center), collapse = "; "),
+"short_histology","Abbreviated `integrated_diagnosis` or `broad_histology` for plotting purposes",paste(unique(histology_df$short_histology), collapse = "; "),
+"tumor_descriptor","Phase of therapy from which tumor was derived",paste(unique(histology_df$tumor_descriptor), collapse = "; "),
+"tumor_fraction","Theta2 tumor DNA fraction estimate","numeric",
+"tumor_ploidy","Control-FREEC ploidy","numeric"
+)
+
+# Sheet 2: Define array of samples that need to be classified from [ticket](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1206) which are not in v21, but set to be changed in final release [v22](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/1207)
```
Comment (sjspielman) [inline tables/output_tables.Rmd:124]: 👍

File: tables/output_tables.Rmd [hunk 6]
```diff
@@ -76,18 +76,86 @@ histology_df %>%
...
   mutate(molecular_subtype = ifelse(
     Kids_First_Biospecimen_ID %in% to_be_classified_samples,
     "MB, To be classified",
     molecular_subtype
-  )) %>%
-  # export with argument `keepNA=TRUE` so NAs come up as NAs and not blank cells
-  write.xlsx(table_s1_file, overwrite=TRUE, keepNA=TRUE)
+  ))
+
+# Sheet 3: CNS region definition based on definitions from [Cassie Kline](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/1025).
```
Comment (sjspielman) [inline tables/output_tables.Rmd:135]: I am not seeing a lot of this information in the attached text document. For example, spine, suprasellar,  "other", Posterior fossa. The other link in this PR to the d3b repo is 404.

Is there somewhere else we have this full information for the samples?

**Comment by sjspielman** [comment]
> just had a question as to whether we want to keep the programmatic option of all cancer groups there since it is quite long. I suppose people can wrap that column, so may be of interest.

I think it's fine to leave this long, since it's excel and people can definitely wrap it.

**Review by sjspielman — CHANGES_REQUESTED** [review]

File: tables/output_tables.Rmd (hunk 6 above)
Comment (jharenza) [reply tables/output_tables.Rmd:135]: Ah, this was the manual pathology report inspection and updating of CNS region, which was separate of the logic - let me find that and add it.

Comment (jharenza) [reply tables/output_tables.Rmd]: updated in last commit with JSON assistance from @logstar

File: tables/output_tables.Rmd [hunk 7]
```diff
@@ -76,18 +76,80 @@ histology_df %>%
...
-  # export with argument `keepNA=TRUE` so NAs come up as NAs and not blank cells
-  write.xlsx(table_s1_file, overwrite=TRUE, keepNA=TRUE)
+  ))
+
+# Sheet 3: CNS region definition based on definitions from [Cassie Kline](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/tables/input/CNS_primary_site_match.json) with additional manual review of [HGG primary_site](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/1025). 
+# This is integrated upstream of molecular subtyping
+
+# CNS_region ~ primary_site matches
+cns_regions_df <- purrr::imap_dfr(jsonlite::fromJSON(CNS_match_json), 
+                                     function(x, name) { tibble::tibble(CNS_region = name, primary_site = paste0(x, collapse = '|')) })
```
Comment (sjspielman) [inline tables/output_tables.Rmd:140]: I'd change `|` to `;` to match the other sheet.

**Review by sjspielman — APPROVED** [review]
Looks good! Thanks for adding in all the sources for the info. This will be pretty easier to track and change if we ever need to :) 

I left one comment about `|` ->`;`, and the Rmd needs to be re-knitted I suspect because I'd expect an HTML diff here.

Assuming all is good with those changes I don't need to see again unless something else needs to change!

File: tables/output_tables.Rmd (hunk 7 above)
Comment (jharenza) [reply tables/output_tables.Rmd:140]: I went back and forth on this - this is more of a dictionary, so it is an "or" operation, whereas the other sheet can have multiple values. But, if it seems like it won't be confusing (some people may not even know what `|` means!), I can use `;`

**Pull request merged**